client/my-sites/plans-features-main/components/plan-upsell-modal/paid-plan-is-required-dialog.tsx

```diff
@@ -41,16 +43,26 @@ export default function PaidPlanIsRequiredDialog( {
 		onFreePlanSelected();
 	}
 
+	const localeSlug = useLocale();
+	const isEnglish = localeSlug && localeSlug.startsWith( 'en' );
+	const upsellDescription =
+		isEnglish ||
```

Comment:
Is it required to check for locale? I thought hasTranslation would cover this scenario as well.

Reply:
Nope, it doesn't cover it. Before doing this check, I still got the old one in an English blog.

Reply:
I have checked it again. With the English check:
<img width="703" alt="Screenshot 2023-12-19 at 11 13 53" src="https://github.com/Automattic/wp-calypso/assets/3832570/b5178b6b-4f6b-40be-8682-56e48aef7d5e">

Without the English check:
<img width="694" alt="Screenshot 2023-12-19 at 11 14 40" src="https://github.com/Automattic/wp-calypso/assets/3832570/233d55e2-6d60-4097-ae39-7be63d9cbde7">

So, yeah, sadly, it doesn't handle the English case. I need to leave the check there.

Reply:
Gotcha, thanks for checking Juan!